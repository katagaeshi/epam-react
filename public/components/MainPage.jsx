import React from 'react';
import RunningTitle from './RunningTitle';
import SearchPanel from './SearchPanel';
import FoundMovies from './FoundMovies';
import ErrorBoundary from './ErrorBoundary';

class MainPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div>
        <ErrorBoundary>
          <RunningTitle />
          <SearchPanel />
          <FoundMovies />
          <RunningTitle />
        </ErrorBoundary>
      </div>
    );
  }
}

export default MainPage;
