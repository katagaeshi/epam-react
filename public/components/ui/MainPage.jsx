import React from 'react';
import PropTypes from 'prop-types';
import RunningTitle from './RunningTitle';
import SearchPanel from './../containers/SearchPanel';
import FoundMovies from './../containers/FoundMovies';
import MovieDetails from './../containers/MovieDetails';
import ErrorBoundary from './ErrorBoundary';

const MainPage = props => (
  <div>
    <ErrorBoundary>
      <RunningTitle />
      {props.activePanel === 'SearchPanel' ? <SearchPanel /> : <MovieDetails /> }
      <FoundMovies />
      <RunningTitle />
    </ErrorBoundary>
  </div>
);

MainPage.propTypes = {
  activePanel: PropTypes.string.isRequired,
};

export default MainPage;
