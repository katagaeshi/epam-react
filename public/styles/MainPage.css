.MainPage {
  display: flex;
  flex-flow: column;
}

.MainPage > * {
  flex: 1 100%;
}
