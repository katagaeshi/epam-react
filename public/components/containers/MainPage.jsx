// @flow

import { connect } from 'react-redux';
import MainPage from '../ui/MainPage';
import {
  setRedirect,
  findMovies,
} from './../../actions';

const mapStateToProps = state => ({
  activePanel: state.activePanel,
  movieDetailsId: state.movieDetails.toString(),
  redirect: state.redirect,
  searchQuery: state.searchQuery,
});

const mapDispatchToProps = dispatch => ({
  setRedirect: value => dispatch(setRedirect(value)),
  findMovies: (query) => {
    dispatch(findMovies({
      query,
      value: null,
      filter: null,
      sortOption: null,
    }));
  },
});

const Container = connect(mapStateToProps, mapDispatchToProps)(MainPage);
export default Container;
