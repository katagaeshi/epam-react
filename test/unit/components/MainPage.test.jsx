import MainPage from 'public/components/MainPage';

describe('MainPage', () => {
  it('Snapshot', () => {
    const tree = shallow(<MainPage />);
    expect(tree).toMatchSnapshot();
  });
});
