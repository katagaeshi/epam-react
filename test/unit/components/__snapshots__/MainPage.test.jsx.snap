// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Snapshot 1`] = `
<div>
  <ErrorBoundary>
    <RunningTitle
      text="netflixroulette"
    />
    <SearchPanel />
    <FoundMovies
      movies={null}
      total="No films found"
    />
    <RunningTitle
      text="netflixroulette"
    />
  </ErrorBoundary>
</div>
`;
